test(ProductiveCard): add avt test coverage

devadula-nandan opened the pull request:
Closes #6472

adds accessibility tests for productive card

Since we are in v12 OverflowMenu. i noticed that passing aria-label prop to v12 OverflowMenu spreads it to the overflow menu container which is a div and is reported as violation
v12 OverflowMenu uses single `Label` prop to set tooltip text on menu button and aria-label for list container (ul)
which means we might need to deprecate either `iconDescription` or `overflowAriaLabel` as `Label` prop handles both with v12 OverflowMenu.

useful references
https://github.com/carbon-design-system/ibm-products/pull/5803
https://github.com/carbon-design-system/ibm-products/pull/6395

i have passed label as below, and awaiting for any review suggestions
```
label={iconDescription || overflowAriaLabel}
```

#### What did you change?
e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js
packages/ibm-products/src/components/Card/Card.tsx

#### How did you test and verify your work?
yarn avt

elycheea commented on e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js:
```diff
@@ -23,4 +24,226 @@ test.describe('ProductiveCard @avt', () => {
+
+    await expect(page).toHaveNoACViolations('ProductiveCard @avt-disabled');
+
+    const buttons = page.locator(`button.${carbon.prefix}--btn`);
+    const disabledButton = buttons.nth(2);
+    expect(await disabledButton.getAttribute('disabled')).not.toBeNull();
```
I think we typically try to query by role or text over a specific query selector.

In this case, we could probably do 

```js
const disabledButton = page.getByRole('button', { name: 'Read more' });
```

Or some similar variation of that.

elycheea commented on e2e/components/ProductiveCard/ProductiveCard-test.avt.e2e.js:
```diff
@@ -23,4 +24,226 @@ test.describe('ProductiveCard @avt', () => {
+      id: 'ibm-products-components-cards-productivecard--with-overflow',
+    });
+
+    const menuButton = page.locator('button[aria-haspopup="true"]');
+    const menu = page.locator('ul[role="menu"]');
```
We can probably use the roles here as well. :)

https://playwright.dev/docs/locators#locate-by-role

elycheea reviewed (commented):
Looks good overall, but I think there are a few places where we could also use role/text locators rather than by query selectors. :)

devadula-nandan commented:
@elycheea fixed it

elycheea reviewed (approved):
Also agree it might be good to deprecate `iconDescription` or `overflowAriaLabel` if they are now redundant. If `iconDescription` overrides `overflowAriaLabel`, that may be preferable unless the latter is more widely used. 🤔

devadula-nandan commented:
> Also agree it might be good to deprecate `iconDescription` or `overflowAriaLabel` if they are now redundant. If `iconDescription` overrides `overflowAriaLabel`, that may be preferable unless the latter is more widely used. 🤔

im pretty sure we can deprecate `iconDescription` as it has been introduced recently compared to `overflowAriaLabel`.
also changed the override order to prioritise `overflowAriaLabel`. and updated the pr accordingly.

although a more generic name `label` could be more convenient (just like we have in core. a common prop to pass to button tooltip and aria-label for v12 OverflowMenu) when compared to having `overflowAriaLabel` that also sets tooltip text.

any additional thoughts or suggestions?

anamikaanu96 reviewed (changes requested):
Hi @devadula-nandan can you add test for hover state.

devadula-nandan commented:
@anamikaanu96 done
